vm/map.c

```diff
@@ -1054,19 +1098,31 @@ void vm_mapGetStats(size_t *allocsz)
 
 vm_map_t *vm_getSharedMap(syspage_prog_t *prog, int map)
 {
+	vm_map_t *ret = NULL;
+
 #ifdef NOMMU
 	unsigned int i;
 
-	if (map < 0)
-		map = prog->dmaps[0];
+	if (prog == NULL) {
+		if (map >= 0 && (size_t)map < map_common.mapssz) {
+			ret = map_common.maps[map];
+		}
+	}
+	else {
+		if (map < 0) {
+			map = prog->dmaps[0];
```

Comment:
TBH I don' t like that here we assigning variable and two lines below checking if it's equal.  I have to read it twice to recognize what was intended.

But it's old code and changing it increase indentation without early return. So approving anyway.